### PR TITLE
Reorganization and improvements to hygdrop

### PR DESCRIPTION
**Pull request by copyninja**
This commit includes improvements and reorganization of hygdrop code. It can be summarized as follows
- Plugin architecture implemented. New feature can be added as hy file under plugins directory. And new hy file should contain function `process(connection, event, message)` which is responsible to process message as required. Hygdrop doesn't know what is plugin does processing is delegated to plugin itself.
- Code is now fixed to work with new changes in hy compiler related area
- Spy feature added to bot so it can print python source of input.
- Core member listing for hylang is implemented
- Any project or repo can be given to hy and it will fetch issue/commit for the same from github (previously it was working only for hylang/hy)
- nose test cases added

Some more planned todo's
- [x] Travis integration
- [ ] ~~pypi integration~~
- [ ] Fun stuff with bot (more commands in same fashion as fsbot on #emacs)
- [x] requirements.txt updating
- [ ] improve test cases
- [x] CLI integration
- [ ] If lines are too long the pastebin the result
- [ ] The errors also should be sent to IRC instead of sys.stderr

### CODE REVIEW

**Comment by berkerpeksag** [comment]
- It would be great to add a CLI to hygdrop.hy (e.g. `hy hygdrop.hy -p 1234 -c #hy, #python`)
- All Python code should follow PEP 8
- All Hy code should follow the [Hy style guide](https://github.com/hylang/hy-style-guide)
- Could you squash the commits when you're done?

**Comment by copyninja** [comment]
Can you tell me what are the hy things I'm not following? other than global variables surrounded by _var_ ? Also squashing pushed commit is not the right thing to do besides other than duplicate README file update commits all there is no unnecessary commit.

File: plugins/codeeval.hy [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  (setv sys.stdout (StringIO))
+  (.runsource *hr* code)
+  (if dry-run
+    (.replace (.getvalue sys.stdout) "\n" " ")
+    (.privmsg connection target (.replace (.getvalue sys.stdout) "\n" " "))))
+
+(defun source-code [connection target hy-code &optional [dry-run False]]
+  (let [[astorcode (-> (import_buffer_to_hst hy-code)
+		       (hy_compile __name__ ))]
+	[pysource (.to_source astor.codegen astorcode)]]
```
Comment (theanalyst) [inline plugins/codeeval.hy:28]: could you indent this to match the other let binding L#26

Comment (copyninja) [inline plugins/codeeval.hy:28]: I don't know what is wrong but indentation seems fine in my commit https://github.com/copyninja/hygdrop/blob/develop/plugins/codeeval.hy#L28.

Comment (theanalyst) [inline plugins/codeeval.hy:28]: alright..seems to be some GH indentation messup then

**Comment by Foxboron** [comment]
I'll do a :+1: and merge if nobody have anything more to add ^^

**Comment by berkerpeksag** [comment]
Thanks @copyninja!

> Can you tell me what are the hy things I'm not following?

There are a couple of 1-space indentations in plugins/codeeval.hy and plugins/github.hy, but we can ignore them :)

**Pull request merged**